src/Neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -153,11 +155,10 @@ protected internal async void NativeOnPersist()
             {
                 if (Trigger != TriggerType.OnPersist)
                     throw new InvalidOperationException();
-                foreach (NativeContract contract in NativeContract.Contracts)
-                {
-                    if (contract.IsActive(ProtocolSettings, PersistingBlock.Index))
-                        await contract.OnPersist(this);
-                }
+                var taskList = NativeContract.Contracts
+                    .Where(w => w.IsActive(ProtocolSettings, PersistingBlock.Index))
+                    .Select(async s => await s.OnPersist(this));
+                await Task.WhenAll(taskList);
```

Comment:
Consider GAS distributions in NEO and Oracle `PostPersist`. Or any other inter-contract state races or dependencies.

Reply:
it's still running sync

Reply:
OK if so. I can only wonder why it'd be faster, but if it is then fine. Still, I'd test these changes (and there is a number of them) separately to ensure each of them makes any difference and worth doing.

Reply:
I'm still working on performance issues with the profiler reports. As of now this would be future use this method. If one got set to inactive _(native contracts)_. I know it doesn't do much of an increase for this one.